tests/test_viewport.py

```diff
@@ -96,7 +96,7 @@ def draw_fn(draw, width, height):
         assert width == 10
 
     sshot = snapshot(10, 10, draw_fn, interval)
-    assert sshot.last_updated == -interval
+    assert sshot.last_updated == 0.0
```

Comment:
The test was correct before, the last updated value should be `-interval` - the code you added was a regression - see other comment :)

Reply:
sorry if there was some confusion over this

Reply:
No, confusion was on my side.  I've hopefully resolved both this and the original unintended change to the snapshot class.  Not sure where that came from...